codec-http/src/main/java/io/netty/handler/codec/http/HttpContentEncoder.java

```diff
@@ -319,4 +327,22 @@ public EmbeddedChannel contentEncoder() {
             return contentEncoder;
         }
     }
+
+    private static final class RequestMetaData {
```

Comment:
Could you give some more detail ? Also we could use the Recycler for this or a custom queue which keeps the RequestMetaData objects.